modules/_labs/combobox/react/lib/Combobox.tsx

```diff
@@ -121,54 +118,85 @@ const listBoxIdPart = `listbox`;
 const optionIdPart = `option`;
 const getOptionId = (baseId?: string, index?: number) => `${baseId}-${optionIdPart}-${index}`;
 
-export default class Combobox extends React.Component<ComboboxProps, ComboboxState> {
-  private inputRef = this.props.children.props.inputRef || React.createRef<HTMLInputElement>();
-  private comboboxRef = React.createRef<HTMLDivElement>();
-  private id = uuid();
-
-  static defaultProps = {
-    clearButtonLabel: `Reset Search Input`,
-    clearButtonVariant: IconButton.Variant.Plain,
-  };
-
-  state: Readonly<ComboboxState> = {
-    isFocused: false,
-    value: '',
-    showingAutocomplete: false,
-    selectedAutocompleteIndex: null,
-  };
-
-  componentDidMount() {
-    if (this.props.initialValue) {
-      this.setInputValue(this.props.initialValue);
+const getTextFromElement = (children?: React.ReactNode) => {
+  let text = '';
+  React.Children.map(children, child => {
+    if (child == null || typeof child === 'boolean' || child === {}) {
+      text += '';
+    } else if (typeof child === 'string' || typeof child === 'number') {
+      text += child.toString();
+    } else if (Array.isArray(child)) {
+      text += getTextFromElement(child);
+    } else if ('props' in child) {
+      text += getTextFromElement(child.props.children);
     }
-  }
-
-  componentDidUpdate(prevProps: ComboboxProps, prevState: ComboboxState) {
-    const listboxOrFocusChanged =
-      this.props.autocompleteItems !== prevProps.autocompleteItems ||
-      this.state.isFocused !== prevState.isFocused ||
-      this.state.value !== prevState.value;
-    if (this.props.autocompleteItems && listboxOrFocusChanged) {
-      this.setState({
-        showingAutocomplete: this.props.autocompleteItems.length > 0 && this.state.isFocused,
-      });
+  });
+  return text;
+};
+
+const buildStatusString = (listCount: number): string => {
+  return `There ${listCount === 1 ? 'is' : 'are'} ${listCount} suggestion${
+    listCount === 1 ? '' : 's'
+  }.`;
+};
+
+const isValidSingleChild = (child: React.ReactNode) => {
+  return React.isValidElement(child) && React.Children.only(child);
+};
+
+const Combobox: React.FC<ComboboxProps> = ({
```

Comment:
Nit: We've been recommending using `const Component = (props: ComponentProps) => {}` over this format for a reason I can't seem to think of at the moment haha. I will look into it and get back to you, but do you mind updating the formatting here so it aligns with our other functional components?

Reply:
Looks like it uncovered a potential bug by typing children.props better

Reply:
```
const Component: React.FC<ComponentProps> = (props) => {
  props.children.props.helicopter // helicopter is typed as any
}
```
```
const Component = (props: ComponentProps) => {
    props.children.props.helicopter // helicopter is typed correctly as Badass
}
```

Reply:
That was why! Thanks for testing it out 👍